src/examples/java/com/amazonaws/crypto/examples/BuilderExample.java

```diff
@@ -0,0 +1,71 @@
+package com.amazonaws.crypto.examples;
```

Comment:
Examples are great however we should prefer unit test cases as they assert the functionality as part of the build and also show the usage of it.

Reply:
Yes, and we should also just update existing examples to use the new API instead of adding a new bespoke example.

Reply:
@bdonlan,
 Since you are the contributor and know more about this project than I do.
I have commented it in other PR and asking the same question here again.
Do you think we should put examples as part of the main sdk library,
or we should either create separate maven modules or keep them under src/test?
Usually the tendency is that being it an example, developers can put less secured code and insensitive logging or other kinds of code which could be prone to security threats. e.g. example providing public methods or utils which could reveal secrets. This is the one example I can think of right now however there could be more.
just a thought.

Reply:
Generally I think if we're checking in example code we should make sure it's _production quality_ example code that we want people to imitate. We should also avoid pulling in new dependencies for examples in the main repository. For the more complex example you refer to, since it involves a lot of moving parts (a client component, a server in lambda, a cloudformation template...) it might be preferable to split it off to avoid confusion - or maybe have it as a separate maven subproject in a subdirectory? I'm not enough of a maven expert to know if that is a good approach here though.

Reply:
agree with you.
I have maven expertise and will create PR so that you can review if it makes sense.

Reply:
created https://github.com/awslabs/aws-encryption-sdk-java/issues/32